cupy/linalg/norms.py

```diff
@@ -76,7 +75,11 @@ def norm(x, ord=None, axis=None, keepdims=False):
             return abs(x).sum(axis=axis, keepdims=keepdims)
         elif ord is None or ord == 2:
             # special case for speedup
-            s = (x.conj() * x).real
+            if x.dtype.kind == 'c':
+                s = abs(x)
+                s *= s
+            else:
+                s = x * x
```

Comment:
I don't immediately see that this is a performance improvements but do you happen to have any benchmarks (or can otherwise explain why)?

Reply:
This form with abs() followed by in-place multiplication was already used in two of the other code branches and was faster when I measured it. I would guess it has to do with needing one fewer temporary array.

For small arrays I get nearly equivalent times:
```Python
import cupy
dev = cupy.cuda.device.Device(0)
x_small = cupy.ones((16, 16), dtype=cupy.complex64)
x_large = cupy.ones((4096, 4096), dtype=cupy.complex64)

%timeit s = abs(x_small); s *= s; dev.synchronize()
#>> 30.2 µs ± 18.4 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)

%timeit s = (x_small.conj() * x_small).real; dev.synchronize()
#>> 29.3 µs ± 12.2 µs per loop (mean ± std. dev. of 7 runs, 1 loop each)
```

but for large arrays it is substantially faster
```Python
%timeit s = abs(x_large); s *= s; dev.synchronize()
#>> 868 µs ± 938 ns per loop (mean ± std. dev. of 7 runs, 1000 loops each)

%timeit s = (x_large.conj() * x_large).real; dev.synchronize()
#>> 1.74 ms ± 533 ns per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```

Reply:
That makes sense, thanks for the explanation.